File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    const answer = [];

    for (const [s, e, k] of queries) {
        const tmp = arr.slice(s, e + 1).filter(x => x > k);

        if (tmp.length === 0) {
            answer.push(-1);
        } else {
            answer.push(Math.min(...tmp));
        }
    }

    return answer;
}

File: 프로그래머스/0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    results = []
    
    for s, e, k in queries:
        min_val = float('inf') 
        
        for i in range(s, e + 1):
            current_value = arr[i]
            
            if current_value > k and current_value < min_val:
                min_val = current_value
                
        if min_val == float('inf'):
            results.append(-1)
        else:
            results.append(min_val)
            
    return results